src/tty.rs

```diff
@@ -68,3 +68,92 @@ pub fn setup_console(console_fd: FileDescriptor) -> Result<()> {
     close(console_fd.as_raw_fd())?;
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    
+    use std::env;
+    use std::fs::{self, File};
+    use std::os::unix::net::UnixListener;
+    use std::path::PathBuf;
+
+    use serial_test::serial;
+    use tempfile::TempDir;
+
+    
+    fn setup() -> Result<(PathBuf, PathBuf, PathBuf)> {
+        let testdir = TempDir::new()?;
+        let testdir_path = testdir.into_path();
+        let rundir_path = Path::join(&testdir_path, "run");
+        let _ = fs::create_dir(&rundir_path)?;
+        let socket_path = Path::new(&rundir_path).join("socket");
+        let _ = File::create(&socket_path);
+        env::set_current_dir(&&testdir_path)?;
+        Ok((testdir_path, rundir_path, socket_path))
+
+    }
+
+    fn teardown(tmpdir: PathBuf) -> Result<()> {
+        fs::remove_dir_all(tmpdir)?;
+        Ok(())
+    }
+
+    #[test]
+    #[serial]
+    fn test_setup_console_socket() {
+        let init =  setup();
+        assert!(init.is_ok());
+        let paths = init.unwrap();
+        let lis = UnixListener::bind(Path::join(&paths.0, "console-socket"));
+        assert!(lis.is_ok());
+        let fd = setup_console_socket(&paths.1, &paths.2);
+        assert!(fd.is_ok());
+        assert_ne!(fd.unwrap().as_raw_fd(), -1);
+        let cleanup = teardown(paths.0);
+        assert!(cleanup.is_ok());
+    }
+
+    #[test]
+    #[serial]
+    fn test_setup_console_socket_empty() {
+        let init =  setup();
+        assert!(init.is_ok());
+        let paths = init.unwrap();
```

Comment:
Other codes, too, but how about something like the following?
```suggestion
        let (testdir_path, rundir_path, socket_path) = init.unwrap();
```

Reply:
Changed it to the following

Reply:
Thank you for reviewing